src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -436,18 +462,26 @@ def _query_workspace_with_load_id_mass(
     load_id: str,
     obj_names: list[str],
     batch_size: int = _WS_MAX_BATCH_SIZE,
-) -> tuple[list[str], list[str]]:
-    uploaded_obj_names = []
-    uploaded_obj_upas = []
-
-    for idx in range(0, len(obj_names), batch_size):
-        obj_names_batch, obj_upas_batch = _query_workspace_with_load_id(
-            ws, workspace_id, load_id, obj_names[idx: idx + batch_size]
-        )
-        uploaded_obj_names.extend(obj_names_batch)
-        uploaded_obj_upas.extend(obj_upas_batch)
-
-    return uploaded_obj_names, uploaded_obj_upas
+    assembly_objs_only: bool = True,
+) -> tuple[list[str], list[str], list[str], list[str]]:
+
+    # TODO: run _query_workspace_with_load_id in parallel instead of in serial
+    (uploaded_obj_names,
+     uploaded_obj_upas,
+     uploaded_assembly_objs_info,
+     uploaded_genome_objs_info) = zip(*[_query_workspace_with_load_id(ws,
+                                                                      workspace_id,
+                                                                      load_id,
+                                                                      obj_names[i:i + batch_size],
+                                                                      assembly_objs_only=assembly_objs_only)
+                                        for i in range(0, len(obj_names), batch_size)])
+
+    return (
+        [item for sublist in uploaded_obj_names for item in sublist],
+        [item for sublist in uploaded_obj_upas for item in sublist],
+        [item for sublist in uploaded_assembly_objs_info for item in sublist],
+        [item for sublist in uploaded_genome_objs_info for item in sublist]
+    )
```

Comment:
Here I think the loop is a lot more readable

Reply:
👍

Reply:
:+1: